vdr/vdr.go

```diff
@@ -147,8 +147,16 @@ func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, _ *t
 		return err
 	}
 
+	controllers, err := r.resolveControllers(*currentDIDDocument)
+	if err != nil {
+		return fmt.Errorf("error while finding controllers for document: %w", err)
+	}
+	if len(controllers) == 0 {
+		return fmt.Errorf("could not find any contollers for document")
+	}
+
 	// TODO: look into the controller of the did for a signing key
-	keyID := currentDIDdocument.Authentication[0].ID.String()
+	keyID := controllers[0].Authentication[0].ID.String()
```

Comment:
always the first?

Reply:
This is a good question. How to choose one? I think we should or just pick one, or specify which controller to use.

Reply:
Or just return an error when there's more ("this is not implemented yet") so we don't accidentally use the wrong one.

I always prefer erroring out over silently doing the wrong thing.

Reply:
https://github.com/nuts-foundation/nuts-specification/issues/39